src/main/tools/build-runfiles.cc

```diff
@@ -310,9 +310,20 @@ class RunfilesCreator {
     }
   }
 
-  FileType DentryToFileType(const std::string &path, char d_type) {
-    if (d_type == DT_UNKNOWN) {
-      struct stat st;
+  FileType DentryToFileType(const std::string &path, struct dirent *ent) {
```

Comment:
While you are no doubt correct here, is there an actual reason for doing this other than the theoretical argument of "POSIX compatibility"?

Reply:
Some context here: https://github.com/bazelbuild/bazel/issues/4719

Reply:
@lberki I'm actually working on the SmartOS (aka Solaris, Aka Illumos) port ;-) in which this is an issue. I have not verified other port targets but this can also be an issue for other (POSIX compliant) port targets like NetBSD. After I'm done with SmartOS my plan is to see if I can help out the NetBSD people.